simple-search-engine-maven/src/test/java/searchengine/model/SearchEngineContextTest.java

```diff
@@ -0,0 +1,219 @@
+package searchengine.model;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import searchengine.service.implementation.SearchEngineAll;
+import searchengine.service.implementation.SearchEngineAny;
+import searchengine.service.implementation.SearchEngineNone;
+
+import java.util.ArrayList;
+import java.util.Set;
+
+class SearchEngineContextTest {
+
+    SearchDataset dataset = new SearchDataset(new ArrayList<>() {
```

Comment:
A good practice is to do this initilization in a setup method (also keep field private): 

```java
...
private SearchDataset dataset;

@BeforeEach
public void setUp(){
   dataset =  new SearchDataset(new ArrayList<>() {....});
}
```
this method will re-initilize the data for each test